src/panels/Menu/TopMenuBar/TopMenuBar.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { Button, Group, Kbd, Menu, Text } from '@mantine/core';
+import { modals } from '@mantine/modals';
+
+import { useGetBoolPropertyValue, useOpenSpaceApi } from '@/api/hooks';
+import {
+  AddFileIcon,
+  ConsoleIcon,
+  ExitAppIcon,
+  SaveIcon,
+  VisibilityIcon
+} from '@/icons/icons';
+import { WindowLayoutOptions } from '@/windowmanagement/WindowLayout/WindowLayout';
+
+import { HelpMenu } from './HelpMenu';
+import { NewWindowMenu } from './NewWindowMenu';
+import { TaskBarMenuChoices } from './TaskBarMenuChoices';
+import { MenuWrapper } from './MenuWrapper';
+
+interface TopMenuBarProps {
+  visibleMenuItems: string[];
+  setVisibleMenuItems: Dispatch<SetStateAction<string[]>>;
+  addWindow: (component: React.JSX.Element, options: WindowLayoutOptions) => void;
+}
+
+export function TopMenuBar({
+  visibleMenuItems,
+  setVisibleMenuItems,
+  addWindow
+}: TopMenuBarProps) {
+  const luaApi = useOpenSpaceApi();
+
+  const [isConsoleVisible, setIsConsoleVisible] =
+    useGetBoolPropertyValue('LuaConsole.IsVisible');
+
+  function toggleLuaConsole() {
+    setIsConsoleVisible(!isConsoleVisible);
+  }
+
+  function toggleShutdown() {
+    return modals.openConfirmModal({
+      title: 'Confirm action',
+      children: <Text>Are you sure you want to quit OpenSpace?</Text>,
+      labels: { confirm: 'Quit', cancel: 'Cancel' },
+      confirmProps: { color: 'red' },
+      onConfirm: () => luaApi?.toggleShutdown()
+    });
+  }
+
+  return (
+    <Group h={30} w={'100%'} bg={'var(--mantine-color-gray-filled)'}>
```

Comment:
Entering design territory but i would prefer the color to be darker hehe

Reply:
I'll postpone it till we do the theme task